presto-main/src/main/java/com/facebook/presto/execution/SqlTaskManager.java

```diff
@@ -345,22 +391,54 @@ public String getTaskInstanceId(TaskId taskId)
     }
 
     @Override
-    public TaskInfo updateTask(Session session, TaskId taskId, Optional<PlanFragment> fragment, List<TaskSource> sources, OutputBuffers outputBuffers, OptionalInt totalPartitions)
+    public TaskInfo updateTask(
+            Session session,
+            TaskId taskId,
+            Optional<PlanFragment> fragment,
+            List<TaskSource> sources,
+            OutputBuffers outputBuffers,
+            Optional<TableWriteInfo> tableWriteInfo)
     {
         requireNonNull(session, "session is null");
         requireNonNull(taskId, "taskId is null");
         requireNonNull(fragment, "fragment is null");
         requireNonNull(sources, "sources is null");
         requireNonNull(outputBuffers, "outputBuffers is null");
 
-        if (resourceOvercommit(session)) {
-            // TODO: This should have been done when the QueryContext was created. However, the session isn't available at that point.
-            queryContexts.getUnchecked(taskId.getQueryId()).setResourceOvercommit();
+        SqlTask sqlTask = tasks.getUnchecked(taskId);
+        QueryContext queryContext = sqlTask.getQueryContext();
+        if (!queryContext.isMemoryLimitsInitialized()) {
+            if (resourceOvercommit(session)) {
+                // TODO: This should have been done when the QueryContext was created. However, the session isn't available at that point.
+                queryContext.setResourceOvercommit();
+            }
+            else {
+                queryContext.setMemoryLimits(
+                        getQueryMaxMemoryPerNode(session),
+                        getQueryMaxTotalMemoryPerNode(session),
+                        getQueryMaxBroadcastMemory(session),
+                        getQueryMaxRevocableMemoryPerNode(session));
+            }
         }
 
-        SqlTask sqlTask = tasks.getUnchecked(taskId);
+        queryContext.setVerboseExceededMemoryLimitErrorsEnabled(isVerboseExceededMemoryLimitErrorsEnabled(session));
+        queryContext.setHeapDumpOnExceededMemoryLimitEnabled(isHeapDumpOnExceededMemoryLimitEnabled(session));
+        String heapDumpFilePath = Paths.get(
+                getHeapDumpFileDirectory(session),
+                format("%s_%s.hprof", session.getQueryId().getId(), taskId.getStageExecutionId().getStageId().getId())).toString();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: &#39;getId&#39; expression. This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 428. Consider formatting this data with &#39;%d&#39; instead.